cmake/Platform/Other/ArduinoSDKSeeker.cmake

```diff
@@ -40,3 +40,59 @@ function(find_arduino_sdk _return_var)
 
 endfunction()
 
+#=============================================================================#
+# Attempts to find the Arduino SDK's bin folder in the host system, searching at known locations.
+# Most installs will have it at SDK/hardware/tools/avr/bin but if nothing is there, it will 
+# attempt to find a folder in PATH containing avr-gcc, hoping that everything else will be there too
+# This is because a bunch of linux distros' package managers install the binaries into /usr/bin
+#       _return_var - Name of variable in parent-scope holding the return value.
+#       Returns - Path to the folder containing Arduino compiler binaries
+#=============================================================================#
+function(find_arduino_sdk_bin _return_var)
+
+    if (DEFINED ENV{ARDUINO_SDK_BIN_PATH})
+        string(REPLACE "\\" "/" unix_style_sdk_bin_path $ENV{ARDUINO_SDK_BIN_PATH})
+        set(${_return_var} "${unix_style_sdk_bin_path}" PARENT_SCOPE)
+    elseif (IS_DIRECTORY "${ARDUINO_SDK_PATH}/hardware/tools/avr/bin")
+        set(${_return_var} "${ARDUINO_SDK_PATH}/hardware/tools/avr/bin" PARENT_SCOPE)
+    else ()
+        # Some systems like the Arch Linux arduino package install binaries to /usr/bin
+        find_program(avr_gcc_location avr-gcc)
+        if (avr_gcc_location)
```

Comment:
This is not enough to know for certain that a program has been found.
This is because CMake will set the variable to `name-NOTFOUND`, where `name` is the name of the variable, in this case `avr_gcc_location`.
To cope with that, I tend to regex-match (using the `MATCHES` keyword inside a conditional) the value of the variable to the "NOTFOUND" string, failing if a match exists.